pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -434,6 +434,16 @@ func (m *kubeGenericRuntimeManager) computePodContainerChanges(pod *v1.Pod, podS
 		ContainersToKill:     make(map[kubecontainer.ContainerID]containerToKillInfo),
 	}
 
+	// Get a ref for Event logs. If this fails, podRef will be nil, but the Eventf handler tolerates this.
+	podRef, err := ref.GetReference(api.Scheme, pod)
+	if err != nil {
+		glog.Errorf("Couldn't make a ref to pod %q: '%v'", format.Pod(pod), err)
+	}
+
+	if sandboxChanged && sandboxID != "" {
+		m.recorder.Eventf(podRef, v1.EventTypeWarning, events.SandboxChanged, "Pod sandbox has changed, containers will be killed and recreated")
```

Comment:
All of them, and no, I don't want to add details about the pod because the event is already linked to the pod.  It "stutters" when the pod mentioned twice in the event, IMO.